app/src/androidTest/java/org/sagebionetworks/research/mpower/RoomScheduledActivityTests.kt

```diff
@@ -0,0 +1,239 @@
+package org.sagebionetworks.research.mpower
+
+import android.arch.persistence.room.Room
+import android.support.test.InstrumentationRegistry
+import android.support.test.runner.AndroidJUnit4
+import junit.framework.Assert.assertEquals
+import junit.framework.Assert.assertNotNull
+import junit.framework.Assert.assertNull
+import junit.framework.Assert.assertTrue
+import org.joda.time.DateTime
+
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.sagebionetworks.bridge.rest.RestUtils
+import org.sagebionetworks.bridge.rest.model.ScheduleStatus
+import org.sagebionetworks.bridge.rest.model.ScheduledActivityListV4
+import org.sagebionetworks.research.mpower.room.ResearchDatabase
+import org.sagebionetworks.research.mpower.room.RoomScheduledActivity
+import org.sagebionetworks.research.mpower.room.RoomScheduledActivityDao
+import org.sagebionetworks.research.mpower.room.RoomTypeConverters
+import java.io.IOException
+import java.nio.charset.Charset
+
+@RunWith(AndroidJUnit4::class)
+class RoomScheduledActivityTests {
```

Comment:
Do we also need test cases for get(date)?

Is it worth having explicit test cases for the two insert methods and the delete method?

Do we want a test with compound activities w/ both schema list and survey list just for completeness?

Reply:
I added a test for get(date).

I would like to test compound activities, but I need JSON for that, I've never made or used one on Bridge.

Also, this code was mostly pushed as a progress update on the work I've done so far to get feedback before I go too deep into this approach.  I'll be adding more unit tests when I add/remove functions from the DAO as I build a new ActivityManager for mPower.

Reply:
Oh and I feel like all the tests indirectly test insert and delete, but I can see the value in testing them directly. Any suggestions on what a proper test would look like for those?